tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c

```diff
@@ -42,10 +42,14 @@ static struct k_sem events[TEST_FLAGS];
 static const struct bt_mesh_test_cfg friend_cfg = {
 	.addr = 0x0001,
 	.dev_key = { 0x01 },
+	.net_idx = 0x03,
+	.net_key = { 0x03 }
```

Comment:
As the `bt_mesh_test_setup` loads the config and provisions the device, for persistence test this needs to be configurable so that it can be changed to different values and checked. This change requires it to be aligned in different tests.

Reply:
The net_idx seems fair enough, but the net_key was already configurable, as seen in test_provision.c. Moving it into this struct forces us to initialize it in every test, even if it's not relevant. IMO, it makes more sense to just add net_idx as a variable as well (or combine the two into a struct, stored the same way as `test_net_key`